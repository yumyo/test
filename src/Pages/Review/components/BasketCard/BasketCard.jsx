import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { removeFromBasket } from '../../actions';

export class BasketCardComponent extends Component {
    getCardCost (price, quantity) {
        let cost = price;
        cost = price * quantity;
        return cost;
    }

    handleClick = (id) =>
        this.props.dispatch(removeFromBasket(id));

    render () {
        const { item: { id, title, price, quantity } } = this.props;
        return (
            <div onClick={this.handleClick.bind(this, { id })}>
                <h3>{title}</h3>
                <p>Cost: Â£{this.getCardCost(price, quantity)}</p>
                <p>Quantity: {quantity}</p>
                <button>Remove from Basket</button>
            </div>
        );
    }
}

BasketCardComponent.propTypes = {
    item: PropTypes.shape({
        id: PropTypes.string,
        title: PropTypes.string,
        price: PropTypes.decimal,
        quantity: PropTypes.number,
    }).isRequired,
};

export default connect()(BasketCardComponent);
