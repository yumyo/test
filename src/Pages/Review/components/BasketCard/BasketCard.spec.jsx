import React from 'react';
import { shallow } from 'enzyme';
import { BasketCardComponent } from './BasketCard.jsx';

describe('<BasketCard />', () => {
    let component;
    const props = {
        dispatch: jest.fn(),
        item: {
            id: '2',
        },
    };

    beforeEach(() => {
        component = shallow(<BasketCardComponent {...props} />);
    });

    it('renders', () => {
        expect(component).toMatchSnapshot();
    });
});
