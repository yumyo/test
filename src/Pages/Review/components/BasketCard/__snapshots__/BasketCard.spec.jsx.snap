// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasketCard /> renders 1`] = `
<div
  onClick={[Function]}
>
  <h3 />
  <p>
    Cost: Â£
    NaN
  </p>
  <p>
    Quantity: 
  </p>
  <button>
    Remove from Basket
  </button>
</div>
`;
